### DISCUSSION TITLE
Eager loading relationships?

### DISCUSSION THREAD

**Discussion by edgrosvenor** [filamentphp/filament#202] [Help]
I might be missing something obvious, but is there an "official" way of eager loading relationships when creating resources?

**Comment by ryangjchandler** [comment]
@edgrosvenor Are you talking about for the relation managers?

**Reply by edgrosvenor** [reply]
@ryangjchandler No. Just a table. I have a list of organizations and each organization has several employees. I want to display the number of employees on the table.

**Reply by ryangjchandler** [reply]
Ah, I see. Somebody else has mentioned this but I'll make a note & add it in as part of the refactor of tables & forms. Definitely a good shout though!

**Reply by kjlloydie** [reply]
Out of curiosity, how does one eager load relationships in relation managers? I have a relation manager and the `BelongsToManyMultiSelect` I have on the form doesn't appear to have the relation loaded, as it's blank on edit.

**Reply by danharrin** [reply]
Hmm, eagre loading doesn't need to happen in forms like that. Are you sure that the `BelongsToManyMultiSelect` is configured properly?

**Reply by kjlloydie** [reply]
It was, kind of! 🥴 I actually figured out what the problem was and it was my own fault, sorry!

I'm doing something funky with mapping multiple Model subclasses to a single table (it's a long story) but like an idiot I forgot, so the relationship pivot table was using the "wrong" class name to store/load the relationships. I switched the RelationManager to use the specialised relationship/subclass and it works perfectly, sorry!

**Reply by underdpt** [reply]
> Ah, I see. Somebody else has mentioned this but I'll make a note & add it in as part of the refactor of tables & forms. Definitely a good shout though!

Hello,

Was this implemented? I'm missing a way to eager load relationships.

**Reply by danharrin** [reply]
You can override the `getTableQuery()` method on the List page and return `parent::getTableQuery()->with(...)`, but usually Filament will eager-load relationships for you if you use relationship columns.

**Reply by underdpt** [reply]
Thanks! I was looking for it on the Resource, my fault!

This is for custom columns or closures that check relations to format cells, p.e.. I was indeed testing with hidden columns as a hack.
